src/desktopvirtualization/azext_desktopvirtualization/generated/custom.py

```diff
@@ -0,0 +1,213 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+import json
+
+
+def desktopvirtualization_workspace_list(client,
+                                         resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name=resource_group_name)
+    return client.list_by_subscription()
+
+
+def desktopvirtualization_workspace_show(client,
+                                         resource_group_name,
+                                         workspace_name):
+    return client.get(resource_group_name=resource_group_name,
+                      workspace_name=workspace_name)
+
+
+def desktopvirtualization_workspace_create(client,
+                                           resource_group_name,
+                                           workspace_name,
+                                           location,
+                                           tags=None,
+                                           description=None,
+                                           friendly_name=None,
+                                           application_group_references=None):
+    return client.create_or_update(resource_group_name=resource_group_name,
+                                   workspace_name=workspace_name,
+                                   tags=tags,
+                                   location=location,
+                                   description=description,
+                                   friendly_name=friendly_name,
+                                   application_group_references=application_group_references)
+
+
+def desktopvirtualization_workspace_update(client,
+                                           resource_group_name,
+                                           workspace_name,
+                                           tags=None,
+                                           description=None,
+                                           friendly_name=None,
+                                           application_group_references=None):
+    if isinstance(tags, str):
+        tags = json.loads(tags)
```

Comment:
this is to support json style tags?

Reply:
we could put it in `tags_type` then. @jiasli @zhoxing-ms to comment.

Reply:
In fact, the tags is declared `tags_type` here [link](https://github.com/Azure/azure-cli-extensions/blob/84cccaa09f6c40635fb35c82ed264117c9be9d93/src/desktopvirtualization/azext_desktopvirtualization/generated/_params.py#L49)
In my opinion, this might not be the right way to deal with tag, because the function `validate_tags` of `tags_type` will convert the input parameter of `--tag` into a dict, which should not appear as a string.
By the way, if you want tag to support JSON input, you can do it at `validate_tags`, because we have a plan to support JSON input for tags of all resources, so as to avoid repeated implementation of this processing logic.